Release notes for release 1.6 (#339)
okteto/docs: src/content/self-hosted/install/releases.mdx

id: releases
---

## 1.6.0
13 March, 2023

### Deprecation Notice
- Support for Kubernetes [1.21](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md) has been deprecated and will be removed next release.

> ### Features

Author: That is part of updating the CLI used in the backend. Do we want to mention explicitly also that? Wouldn't be enough with the link to the release notes of `2.13.0`?

> ### Features

Author: That is part of updating the CLI used in the backend. Do we want to mention explicitly also that? Wouldn't be enough with the link to the release notes of `2.13.0`?

> ### Features

Reviewer: No, let's mention the feature explicitly.  Are there any other major CLI features? If so, we should also mention them here since they enable new scenarios on Okteto Self-Hosted.

> ### Features

Author: Not sure in this case if it has any impact. I mean, any pipeline deployed in the okteto instance is already executed remotely, so not sure it worths mentioning in this release notes

> ### Features

Author: ok. The other big feature is the remote deploy and destroy but not sure it worths mentioning here as everything deployed through the UI is already done remotely, so from a user perspective might be confusing to mentioning here. The real difference happens when you run `okteto deploy` locally

> ### Features

Reviewer: I still think that we should mention it. But it's not a blocker. I'll ask Pablo and Matt to follow up on this.

> ### Features

Author: 👍 is there anything else blocking the PR? If not, we could merge it and finish the release process

> ### Features

Reviewer: No, merge and release.

Please don't forget to make 1.6 the default version once 1.6 is available 👍🏽

- Add support for Kubernetes [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md)
- Update the Okteto CLI version to [2.13.0](https://github.com/okteto/okteto/releases/tag/2.13.0)

> - Default ingress class for ingresses managed by okteto is `okteto-nginx`. All ingresses managed by okteto are migrated automatically during the upgrade

Reviewer: "All ingresses managed by Okteto will be automatically updated. No action is required from the user"

Also, can we put this as a breaking change?

Author: I'll update it. Not sure it's a breaking change as we are migrating things to prevent any issue. In a breaking change I would expect some action to be taken from the user

Author: I've removed it from here and added a paragraph at the beginning. Related to this comment: https://github.com/okteto/docs/pull/339/files#r1133865958

- Support [custom quotas](self-hosted/administration/configuration.mdx#quotas) for node port and load balancer service types

> - Public override support for self-signed certificates and bring your own certificate

Author: The idea is to include a link to the documentation related to this PR: https://github.com/okteto/docs/pull/340

> - Wake operation takes into account dependencies between dev environments to wake dependant environments first

Author: Wake things up in order, so a service doesn't wake up before any of its dependencies. How would you state it?

### Improvements

- Several improvements in the UI to keep the status updated accordingly with the status in user's dev environments
- Allow users to access to namespaces that are being terminated to see the operation logs
- Added input autofocus in several dialogs

> - Improve experience in "New namespace" dialog to enable faster the button

Author: The create button was taking some time to be enabled after you type the namespace name. I wasn't sure how to better express it. Open for suggestions

- Filter branches results by search term in GitHub deploy dialog

> - Remove redundant headers in "Settings" view

Reviewer: Which headers?

Author: I'm not sure how to explain it. They were the headers within each admin section. In the screenshot below, the `Secrets` header. 

![Admin section headers](https://user-images.githubusercontent.com/42914013/219382824-aadfc9b7-cc73-4c69-8910-abd0df837fe8.png)

> - Do not autodestroy GC job when it ends

Reviewer: When will it be destroyed? Why is this useful?

Author: Added information

- Private endpoints cookie is now set in the backend and specified as "httpOnly"

> - GC deletes pods stuck in `Failed` phase with `NodeAffinity` reason

Reviewer: This sounds more like a bugfix

Author: Changed

- Upgrade Reloader dependency to [1.0.5](https://github.com/stakater/Reloader/releases/tag/v1.0.5)

### Bugfixes

- Skip volume snapshots deletion as part of destroy all operation if "include volumes" check is not selected
- Fix "Destroy all", "Delete namespace" and "Wake namespace" operations when using private CAs or self-signed certificates
- Display the correct operation name in logs when deleting a namespace or destroying all resources within a namespace
- Avoid duplicated log lines when the browser visibility changes

> - Self-signed certificate is recreated when subdomain or public override changes after an upgrade

Reviewer: If we are talking about the property, use publicOverride. And can we link to the corresponding config logs?

Author: Sure, changed

- Fixed view transition when a destroy namespace operation is retried

> - Fixed missing custom icons on external resources

Reviewer: Can we link to docs?

Author: Sure, changed

id: releases
---

## 1.6.0
13 March, 2023

### Deprecation Notice
- Support for Kubernetes [1.21](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md) has been deprecated and will be removed next release.

### Features

> - Update okteto cli version to [2.13.0](https://github.com/okteto/okteto/releases/tag/2.13.0)

Reviewer 2: ```suggestion
- Update the Okteto CLI version to [2.13.0](https://github.com/okteto/okteto/releases/tag/2.13.0)
```

> - Default ingress class for ingresses managed by okteto is `okteto-nginx`. All ingresses managed by okteto are migrated automatically during the upgrade

Reviewer 2: I don't think we need more docs for this.
@jpf-okteto maybe here, instead of the sameple we have, we can say what's our default nginx configuration (the whole yaml):
https://www.okteto.com/docs/self-hosted/administration/configuration/#nginx-ingress

Author: Done in this PR: https://github.com/okteto/docs/pull/343

- Support [custom quotas](self-hosted/administration/configuration.mdx#quotas) for node port and load balancer service types
- Public override support for self-signed certificates and bring your own certificate

### Improvements

> - Add support for Kubernetes [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md)

Reviewer 2: feature?

Author: When we included `1.23` it was added as improvement that's why I added it as improvement. https://www.okteto.com/docs/self-hosted/install/releases/#improvements-4. No problem on moving it to feature (I think it already makes more sense)

Author: Ok, makes sense, moved to features

- Several improvements in the UI to keep the status updated accordingly with the status in user's dev environments
- Allow users to access to namespaces that are being terminated to see the operation logs

> - Wake operation takes into account dependencies between dev environments to wake dependant environments first

Reviewer 2: I think this is a feature

Reviewer 2: I think it's a feature because it enables new scenarios

Author: Ok, moved to features

- Added input autofocus in several dialogs
- Improve experience in "New namespace" dialog to enable faster the button

> - Filter branches results by search term in GitHub deploy dialog

Reviewer 2: ok, makes sense

id: releases
---

## 1.6.0
13 March, 2023

This version comes with a breaking change. The default ingress class for ingresses managed by okteto is `okteto-nginx` instead of `nginx` to prevent colissions with other controllers that might be already installed in the cluster. All ingresses managed by Okteto will be automatically updated. No action is required from the user. If you want to keep the previous value (`nginx`), you would need to set the keys `ingress.oktetoIngressClass` and `ingress.class` to `nginx` in your helm values file before upgrading Okteto.

In some cases, there is a race condition recreating the controller pods in which a controller pod starts before the new ingress class is created in the cluster and that generates 404 errors when accessing Okteto or dev environments. If that happens, a rollout of the controller deployment fixes the issue (`kubectl rollout restart deployment <ingress-controller-deployment-name> -n okteto`).

> 

Author: @pchico83 @rberrelleza Added this paragraph to explain the ingress class change as breaking change and possible implications. Could you please review it?

id: releases
---

## 1.6.0
13 March, 2023

> This version comes with a breaking change. The default ingress class for ingresses managed by okteto is `okteto-nginx` instead of `nginx` to prevent colissions with other controllers that might be already installed in the cluster. All ingresses managed by Okteto will be automatically updated. No action is required from the user. If you want to keep the previous value (`nginx`), you would need to set the following values in your helm values file:

Reviewer: We use a header for breaking changes, check the older release notes, I think we've done it before.

Author: In `1.0.0` we didn't include it but in a previous one we did so, so I wasn't sure about it. As the most recent one was `1.0.0`, I didn't include it but I think it's better to include it, I'll do it.

> This version comes with a breaking change. The default ingress class for ingresses managed by okteto is `okteto-nginx` instead of `nginx` to prevent colissions with other controllers that might be already installed in the cluster. All ingresses managed by Okteto will be automatically updated. No action is required from the user. If you want to keep the previous value (`nginx`), you would need to set the following values in your helm values file:

Reviewer: ```suggestion
This version comes with a breaking change. The default ingress class for ingresses managed by okteto changed from `nginx` to `okteto-nginx`  to prevent collisions with other controllers that might be already installed in the cluster. There is no action required from the user,  all ingresses managed by Okteto will be automatically updated as part of the upgrade.

If you want to continue to use the previous value (`nginx`), you can do it by setting the following values in your helm values file:
```

```yaml
ingress:
  class: nginx
  oktetoIngressClass: nginx
ingress-nginx:
  controller:
    ingressClass: nginx
    ingressClassResource:
      name: nginx
      controllerValue: "k8s.io/ingress-nginx"
```

> In some cases, there is a race condition recreating the controller pods in which a controller pod starts before the new ingress class is created in the cluster and that generates 404 errors when accessing Okteto or dev environments. If that happens, a rollout of the controller deployment fixes the issue (`kubectl rollout restart deployment <ingress-controller-deployment-name> -n okteto`).

Author: We don't have more information on how frequent it might happen. In fact, it might happen in a fresh installation (I got the error in a fresh installation for `1.5`). In this upgrade might happens also because the ingress class changes, so this will create a new ingress class in the cluster.

I'll rewrite a bit the paragraph to recommend always to rollout the controller to prevent this issue

### Deprecation Notice
- Support for Kubernetes [1.21](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md) has been deprecated and will be removed next release.

### Features

- Add support for Kubernetes [1.24](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md)
- Update the Okteto CLI version to [2.13.0](https://github.com/okteto/okteto/releases/tag/2.13.0)
- Support [custom quotas](self-hosted/administration/configuration.mdx#quotas) for node port and load balancer service types
- Public override support for self-signed certificates and bring your own certificate

> - Wake operation takes into account dependencies between dev environments to wake dependant environments first

Reviewer: ```suggestion
- The wake operation will now wake up dependent services first, following the same order as when the environment was created. 
```

### Improvements

- Several improvements in the UI to keep the status updated accordingly with the status in user's dev environments
- Allow users to access to namespaces that are being terminated to see the operation logs
- Added input autofocus in several dialogs

> - Improve experience in "New namespace" dialog to enable faster the button

Reviewer: ```suggestion
- The "Create Namespace" button is now enabled immediately after you start writing the name of the namespace in the "New Namespace" dialog.
```

> - Filter branches results by search term in GitHub deploy dialog

Reviewer: ```suggestion
- You can now filter the available branches in the GitHub deploy dialog
```
